docs/interactive/launchbuttons.ipynb

```diff
@@ -19,7 +19,24 @@
     "  url                       : https://github.com/yourusername/yourbookrepo  # Online location of your book\n",
     "  path_to_book              : path/to/book  # Optional path to your book, relative to the repository root\n",
     "  branch                    : master  # Which branch of the repository should be used when creating links (optional)\n",
-    "```"
+    "```\n",
+    "The launch button will use [nbgitpuller](https://github.com/jupyterhub/nbgitpuller) to pull your\n",
+    "repository to the chosen Binder, Google Colab, or JupyterHub location.  The `repository:`\n",
+    "configuration line uses `url`, `path_to_book`, and `branch` to find your notebook files _both MyST\n",
+    "Markdown and *.ipynb_.  \n",
+    "\n",
+    "- The `url` should be a GitHub repository that includes all of your files used to build the\n",
+    "  Jupyter-book. Your repository should include:\n",
+    "    - `_build`\n",
+    "    - `_config.yml`\n",
+    "    - `conf.py`\n",
+    "    - `requirements.txt`\n",
+    "    - _any other *.ipynb and *.md files for the book_\n",
+    "- The `path_to_book` should be left blank if the repository is hosting the Jupyter-book. If you are\n",
+    "  using Jupyter-book for documentation, then you may require a `docs/`-path\n",
+    "- The `branch` should be either `master` or `main` unless you have a different development branch\n",
```

Comment:
How about "The `branch` should be the branch where your book's **source files** are stored (usually, `master/` or `main/`)"

Reply:
Here, __source files__ is a great clarifying description. I think it would help to describe source-vs-build to help clarify this distinction.

Reply:
I like the idea of adding a section that differentiates `source/` vs. `build/` files...where do you think is a natural place for this in the docs?

Reply:
> I like the idea of adding a section that differentiates `source/` vs. `build/` files...where do you think is a natural place for this in the docs?

I think it would fit nicely into the subsection in [Publish your book online](https://jupyterbook.org/start/publish.html). Just before diving into the "create an online repository". 

It could serve to define/differentiate some source vs build vocab used in the rest of the book. 

Depends upon how involved the distinction becomes. It might be a short description like you did here and then a link for reference on the actual hierarchy, structure, and content of the build files.

Reply:
If you wanna include a short section or "note" call-out about this I'd be happy to review in this PR, but also fine ifthats better in a new pr

Reply:
> If you wanna include a short section or "note" call-out about this I'd be happy to review in this PR, but also fine ifthats better in a new pr

Happy to add the section on this PR. I can add the clarification you gave me, but you may have to fill in some clarifying details there. 

My plan would be:

* Define source files
* Define build files as the ones in the _build folder
* Tell user to include source, but not build in the github repo

Reply:
Yep that sounds correct to me - it could just be an admonition like:

````
:::{admonition,tip} Source files vs. build files
<content>
:::
````

or something like that?

Reply:
@choldgraf see what you think about the latest commit, 1cf58725086ea0f285947a37648450724d2a7d66

Here is the added content:
## Source vs build files

At this point, you have created a combination of Jupyter notebooks,
markdown files, and configuration files, including `_toc.yml` and
`_config.yml`. These files are your __source__ files. The __source__
files comprise all of the content and configuration that Jupyter Book
needs to build your book.

The `_build` folder contains all of your static website __build__ files.
The __build__ files contain all of the output from Jupyter Book's
`build` command. These files are only used to view your content in a
browser.

Best practice for publishing your book is to keep two branches in your
git project:

- Your book's __source__ files that you create to build the book
- Your book's __build__ files that Jupyter Book created to view the book